### DISCUSSION TITLE
nodejs-mobile cordova plugin

### DISCUSSION THREAD

**Discussion by tobias-klein** [nodejs-mobile/nodejs-mobile#14] [General]
Hi @staltz!

I was happy to see that the nodejs-mobile library was upgraded to Node 16 recently!

Within the [Cordova variant of Ezra Bible App](https://github.com/ezra-bible-app/ezra-bible-app-cordova), I am currently using the [nodejs-mobile-cordova](https://www.npmjs.com/package/nodejs-mobile-cordova) plugin, which was last made available by Janea systems.

What would be my transition path to upgrade to the new version of nodejs-mobile?
Are you planning to publish a new package for the Cordova plugin on npmjs? Or can I directly use this GitHub repo as a source somehow?

Thanks in advance! I appreciate your efforts on this project very much.

**Comment by staltz** [comment]
Hi! We need a maintainer for that plugin. I don't use that module and my knowledge of Cordova is very thin so it's not worth my time to try.

The binaries of nodejs-mobile are the most important piece, and they can be copied to nodejs-mobile-react-native and nodejs-mobile-cordova. These two projects then have React Native specific scripts and Cordova specific scripts to glue everything together.

**Reply by tobias-klein** [reply]
Ok. Well - I would definitely have an interest in making the Cordova plugin work with the latest nodejs-mobile version. I suppose this is the repo where I find the latest sources:
https://github.com/nodejs-mobile/nodejs-mobile-cordova

I just had a look at the structure. It seems like the nodejs-mobile binaries are integrated to the cordova plugin by placing them under `libs/android/libnode`.

So in order to update `nodejs-mobile-cordova` I would have to take the `nodejs-mobile` binaries from the latest release you just published and put them into the `nodejs-mobile-cordova` Git repo?!

Is there anything else you would suggest me to consider?

**Reply by okhiroyuki** [reply]
Hi, I am also using nodejs-mobile-cordova on Android. I'm just thinking of updating my Android, so I'll share any progress I make.

**Reply by tobias-klein** [reply]
I gave it a try to try to build a new version of the nodejs-mobile-cordova plugin based on latest versions of JDK, Android Studio, Gradle, Cordova etc. It failed to build ... I didn't get an app with a functional nodejs plugin, the relevant files were somenow not included in the generated APK.

As of now I am still working with Cordova 7, nodejs-mobile-cordova 0.4.2 and that works. The Cordova build process seems quite complex and if something fails it is not so easy to find a solution.

I think at this point I will continue with what's working for me (nodejs-mobile-cordova 0.4.2) until it stops working due to some (future) Google platform constraints. Whenever I cannot release APKs built with that version on the PlayStore anymore I'll have enough motivation to make the plugin work again and go into debugging/reverse-engineering it.

**Reply by okhiroyuki** [reply]
@tobias-klein 
In the meantime, I have confirmed that node 16.17.1 works on cordova@10.
~~Note that I still need to solve the problem of the missing part of libnode.so in x86. (I still don't quite understand this solution)~~ 
I have figured out how to exclude x86 from the build target.

And it has only been tested on Android.

logcat:

```
2023-02-26 12:39:39.754 22944-23625/*** D/NODEJS-CORDOVA: pluginInitialize
2023-02-26 12:39:39.756 22944-23625/*** D/PluginManager: getPlugin - put: NodeJS
2023-02-26 12:39:39.756 22944-23625/*** V/NODEJS-CORDOVA: setAllChannelsListener

2023-02-26 12:39:39.836 22944-23638/*** D/NODEJS-CORDOVA: Copying node project assets using the files list

2023-02-26 12:39:44.970 22944-23736/*** I/NODEJS-MOBILE: 26 Feb 12:39:44 - [info] Node.js  version: v16.17.1
```

**Reply by tobias-klein** [reply]
@okhiroyuki Cool! Can I find your configuration somewhere on GitHub?

**Reply by okhiroyuki** [reply]
@tobias-klein 
This PR.
https://github.com/okhiroyuki/nodejs-mobile-cordova/pull/13
